### ISSUE TITLE
ember generate docs - specifically model generator

### ISSUE THREAD

**Issue by jgn** [ember-cli/ember-cli#2553]
So this works:

```
ember generate model todo description:string
```

How can one see documentation of the fact that `ember generate model todo` takes additional arguments defining the attributes? `ember help generate model` just provides the list of generators.

**Comment by jakecraige** [comment]
@trabus this something you are still planning on tackling at some point? I'm just running through a few older issues

**Comment by trabus** [comment]
I started work on it, but wasn't quite satisfied with where it was going. There is some refactoring that needs to happen [here](https://github.com/ember-cli/ember-cli/blob/master/lib/cli/lookup-command.js#L24) that I wanted to think through a little more. The way the logic currently works, it doesn't allow for much flexibility, and loses the original command options (which makes looking up specific blueprint help impossible). I'd like to find a way for a command to handle it's own help command call, rather than having the logic in `lookup-command.js`.

We could have something where the help option breaks out of the command, and either calls it's own `printDetailedHelp`, or calls the help command itself. Then we're not slicing args to get the name of the command to lookup, and we can have more control over the options passed.

Anyhow, yes, I did sort of start work on it, but haven't touched it in a couple weeks, and would be fine throwing away what i currently have if someone else wanted to work on it.

**Comment by trabus** [comment]
So I've picked this back up again, and have made considerable progress on extracting the help logic from `lookup-command.js`. I've added a then in `cli.js` which will allow a command to exit before running if the `-h` or `--help` option is present. From there the help command is run with the original options passed in (in addition to the command name).

There are two options in regards to how the `--help` option is added to commands, one that would list it with `availableOptions` for help output, and one that would make it present, but not listed with the help output. Both work essentially the same, it's more just a matter of whether we should list the `--help` option as an availableOption for every command. For example:
```
// with --help added to availableOptions
ember version <options...>
  outputs ember-cli version
  aliases: v, version, -v, --version
  --verbose (Boolean) (Default: false)
  --help (Boolean) (Default: false)
    aliases: -h

// without
ember version <options...>
  outputs ember-cli version
  aliases: v, version, -v, --version
  --verbose (Boolean) (Default: false)
```

Finally, to make this specific issue possible, we'll either need to change the way the help command works, or make exceptions for multi-argument commands like generate and destroy. Currently the help command [will print out help for each param passed to it](https://github.com/ember-cli/ember-cli/blob/master/lib/commands/help.js#L74). I question how frequently this undocumented feature of the help command is used for looking up multiple commands at once. Maybe someone that has access to the metrics might be able to answer? My guess is that it's likely very low, if at all. 

I'm thinking we should just pass all arguments (after the first, which defines the command to lookup) to the command's `printBasicHelp` or `printDetailedHelp` method for it to determine what to do with the additional arguments. This would allow printing help for subcommands, like specific blueprints. The other option would to put in a special case for generate and destroy, where we pass all arguments as previously described.
